test/client/query-subscribe.js

```diff
@@ -441,6 +442,48 @@ module.exports = function(options) {
       connection.get('dogs', 'fido').on('error', done).create({age: 3});
     });
 
+    describe('passing agent.custom to the DB adapter', function() {
+      var connection;
+      var expectedArg = {
+        agentCustom: {foo: 'bar'}
+      };
+      beforeEach('set up', function() {
+        connection = this.backend.connect();
+        connection.agent.custom = {
+          foo: 'bar'
+        };
+      });
+
+      it('sends agentCustom to the db\'s getSnapshot call', function(done) {
+        var query = connection.createSubscribeQuery('dogs', this.matchAllDbQuery);
+        var getSnapshotSpy = sinon.spy(this.backend.db, 'getSnapshot');
+
+        query.on('insert', function() {
+          expect(getSnapshotSpy.callCount).to.equal(2);
+          expect(getSnapshotSpy.getCall(1).args[3]).to.deep.equal(expectedArg);
```

Comment:
Maybe we can also test the first (index 0) call from the `create`, which should also have the `agentCustom`. And/or add a comment explaining which call comes from which